### PR TITLE
[ENGA3-431] : Code Coverage Integration with Sonarcloud

### PR DESCRIPTION
**Pull request by ajzkk**
#### 1. Objective

Code Coverage Integration with Sonarcloud

[ENGA3-431](https://opn-ooo.atlassian.net/browse/ENGA3-431)

#### 2. Description of change

Update pipeline to generate code coverage and sent report to sonarcloud

#### 3. Quality assurance

[Report link (3% after merge)](https://sonarcloud.io/summary/new_code?id=omise_omise-woocommerce&branch=feature%2FENGA3-431)

#### 4. Impact of the change

N/A

#### 5. Priority of change

Normal

#### 6. Additional Notes

N/A

### CODE REVIEW

File: .github/workflows/code-coverage.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          fetch-depth: 0
+
+      - name: Setup PHP with Xdebug
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: "8.1"
+          coverage: xdebug
+
+      - name: Create Folder and File for Vendor Path
+        run: mkdir -p app/etc && cd app/etc && touch vendor_path.php
```
Comment (aashishgurung) [inline .github/workflows/code-coverage.yml:28]: Do we need this for WooCommerce? I think these are specific to Magento.

Comment (ajzkk) [reply .github/workflows/code-coverage.yml:28]: Ahh No need. I will remove it. I over copy.

**Comment by aashishgurung** [comment]
The code coverage workflow works for now but it will not work in the future when we begin testing files containing Wordpress and WooCommerce's core functions like `add_action` and others.

We also did a few tweaks to setup for Magento unit test workflow. Running the unit test with just plugin code didn't work because it didn't found the core Magento classes. We solved it by adding packages that installs Mangeto code for testing. Setting up unit test in WP is a bit more complicated than Magento.

For reference:
- https://github.com/omise/omise-magento/pull/364/files
- https://wpmudev.com/blog/unit-testing-wordpress-plugins-phpunit/
- https://www.codetab.org/tutorial/wordpress-plugin-development/unit-test/plugin-unit-testing/

**Comment by ajzkk** [comment]
> The code coverage workflow works for now but it will not work in the future when we begin testing files containing Wordpress and WooCommerce's core functions like `add_action` and others.
> 
> We did a few tweaks to setup for Magento unit test workflow due to similar reasons. Running the unit test with just plugin code didn't work because it didn't found the core Magento classes. We solved it by adding packages that installs Mangeto code while running Github action to test. Setting up unit test in WP is a bit more complicated than Magento.
> 
> For reference:
> 
> * https://github.com/omise/omise-magento/pull/364/files
> * https://wpmudev.com/blog/unit-testing-wordpress-plugins-phpunit/
> * https://www.codetab.org/tutorial/wordpress-plugin-development/unit-test/plugin-unit-testing/

Yes, I am aware of that and this ticket is just for integration with sonarcloud. We can have new ticket for proper unit test.

**Comment by aashishgurung** [comment]
It's alright. Moving forward, lets start integrating with Sonarcloud only if we have proper unit test setup first.

**Review by aashishgurung — APPROVED** [review]

**Pull request merged**